main.tf

```diff
@@ -1,13 +1,358 @@
+    resources = [aws_secretsmanager_secret.db_secret[each.key].arn]
+    effect    = "Allow"
+  }
+  statement {
+    actions   = ["secretsmanager:ListSecrets", "secretsmanager:ListSecrets"]
+    resources = ["*"]
+    effect    = "Allow"
+  }
+}
+
+resource "aws_iam_user_policy" "secret" {
+  for_each = var.databases   
+  name_prefix   = format("%s-secret-%s", module.aws_user.user_name, each.key)
+  user   = module.aws_user.user_name
+  policy = join("", data.aws_iam_policy_document.secret[each.key].*.json)
+}
+
+
+
+ #########################################################################
+ # Create user policy for the state backend(s3 + dynamodb), 
+ # the user resource s3 bucket
+ #########################################################################
+data "aws_iam_policy_document" "bucket" {
   count = module.this.enabled ? 1 : 0
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  statement {
+    actions   = ["s3:GetObject", "s3:ListBucket", "s3:GetBucketLocation"]
+    resources =  [ format("%s/*", module.terraform_state_backend.s3_bucket_arn), module.terraform_state_backend.s3_bucket_arn]
+    effect    = "Allow"
   }
+  statement {
+    actions   = ["s3:GetObject", "s3:ListBucket", "s3:GetBucketLocation"]
+    resources =  [ format("%s/*", module.s3_bucket.bucket_arn), module.s3_bucket.bucket_arn]
+    effect    = "Allow"
+  }
+   
+  statement {
+    actions   = ["dynamodb:GetItem", "dynamodb:PutItem", "dynamodb:DeleteItem" ]
+    resources =  [ module.terraform_state_backend.dynamodb_table_arn ]
+  }    
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+resource "aws_iam_user_policy" "s3" {
+  count  = module.this.enabled ? 1 : 0
+  name   = module.aws_user.user_name
+  user   = module.aws_user.user_name
+  policy = join("", data.aws_iam_policy_document.bucket.*.json)
+}
+
+
+
+ #########################################################################
+ # Deploy Ingress for BioAnalyze app.
+ #########################################################################
+module "bioanalyze_ingress" {
+  count                   = local.ingress_enabled
+  source                  = "dabble-of-devops-bioanalyze/eks-bitnami-nginx-ingress/aws"
+  version                 = ">= 0.1.0"
+  letsencrypt_email       = var.letsencrypt_email
+  helm_release_values_dir = var.helm_release_values_dir
+  helm_release_name       = var.helm_release_name_ingress
+}
+
+ #########################################################################
+ # Get Ingress for BioAnalyze app.
+ #########################################################################
+data "kubernetes_service" "bioanalyze_ingress" {
+  count = var.enable_ssl == true ? 1 : 0
+  depends_on = [
+    module.bioanalyze_ingress
+  ]
+  metadata {
+    name = "${var.helm_release_name_ingress}-ingress-nginx-ingress-controller"
```

Comment:
k8s namespace for client.

Reply:
do you plan to use one ingress controller per client? I propose to set ingress_namespace here and have the ability to use the same ingress controller per multiple clients. That is the reason I have  helm_release_name_ingress not helm_release_name here.

Reply:
That's a good idea. One ingress controller for all clients is fine.